workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -1,32 +1,87 @@
 ﻿using Improbable.Gdk.Core;
-using Improbable.Gdk.Mobile;
 using UnityEngine;
-using UnityEngine.UI;
 
 namespace Playground
 {
+    [ExecuteInEditMode]
     public class ConnectionScreenController : MonoBehaviour
     {
         private const string HostIpPlayerPrefsKey = "SpatialOSHostIp";
+        private const int GuiPadding = 10;
+        private const float VerticalPositionRatio = 0.4f;
 
-        [SerializeField] private GameObject connectionPanel;
-        [SerializeField] private InputField ipAddressInput;
-        [SerializeField] private Button connectButton;
-        [SerializeField] private Text errorMessage;
         [SerializeField] private GameObject clientWorkerPrefab;
+        [SerializeField] private Font font;
+        [SerializeField] private float screenWidthFontRatio = 20;
 
-        private GameObject worker;
+        private bool showConnectionGui = true;
+        private bool isConnecting = false;
+        private string ipAddressText = null;
```

Comment:
the above two are assigned default values. no need to assign ;)

Reply:
Yes, will clean up

Reply:
Cleaned